app/controllers/bales_controller.rb

```diff
@@ -16,6 +16,11 @@ def show
     render json: bale
   end
 
+  def show_by_material
+    return render_error(1, 'Material must be: "Trash", "Glass" or "Plastic".') unless check_entry
+    render json: Bale.where(material: request.headers['material'])
```

Comment:
Something i've just realized, `material` should be a body param and not a request headers.